### PR TITLE
Update some egressnetworkpolicy case to be compatiable with OVN 

### PR DESCRIPTION
**Pull request by huiran0826**
Update some egressnetworkpolicy cases to be compatible with OVN egressfirewall
case:OCP-11639
SDN logs:https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3/133929/console
OVN logs:https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3/133871/console

@anuragthehatter @zhaozhanqi @weliang1 @rbbratta

### CODE REVIEW

File: features/step_definitions/networking.rb [hunk 1]
```diff
@@ -1122,4 +1122,40 @@
     step %Q/the outputs should contain "sctp"/
     $i +=1
   end
-end
\ No newline at end of file
+end
+
+Given /^get egress data file directory to the#{OPT_SYM} clipboard$/ do | cb_name |
```
Comment (pruan-rht) [inline features/step_definitions/networking.rb:1127]: this step's wording doesn't sound correct to me.  Perhap add `I` or `admin` to the beginning

File: features/step_definitions/networking.rb [hunk 2]
```diff
@@ -1122,4 +1122,40 @@
...
+    cb[cb_name]="ovn-egressfirewall"
+  when "OpenShiftSDN"
+    cb[cb_name]="egressnetworkpolicy"
+  else
+    raise "unknown networkType"
+  end
+  logger.info "The egressfirewall fie directory path is stored to the #{cb_name} clipboard."
+end
+
+Given /^get egress type to the#{OPT_SYM} clipboard$/ do | cb_name |
```
Comment (pruan-rht) [inline features/step_definitions/networking.rb:1145]: this step's wording doesn't sound correct to me. Perhap add `I` or `admin` to the beginning

File: features/step_definitions/networking.rb [hunk 3]
```diff
@@ -1122,4 +1122,40 @@
...
+  ensure_admin_tagged
+  cb_name = "cb_egress_type" unless cb_name
+  @result = admin.cli_exec(:get, resource: "network.operator", output: "jsonpath={.items[*].spec.defaultNetwork.type}")
```
Comment (pruan-rht) [inline features/step_definitions/networking.rb:1148]: @huiran0826 I would recommend adding the class network.operator into `lib/openshift/`

File: features/step_definitions/networking.rb [hunk 4]
```diff
@@ -1122,4 +1122,40 @@
...
     $i +=1
   end
-end
\ No newline at end of file
+end
+
+Given /^get egress data file directory to the#{OPT_SYM} clipboard$/ do | cb_name |
+  ensure_admin_tagged
+  cb_name = "cb_egress_directory" unless cb_name
+  @result = admin.cli_exec(:get, resource: "network.operator", output: "jsonpath={.items[*].spec.defaultNetwork.type}")
```
Comment (pruan-rht) [inline features/step_definitions/networking.rb:1130]: @huiran0826 I would recommend adding the class network.operator into lib/openshift/ so we can get the information in a more object-oriented way.

File: features/step_definitions/networking.rb (hunk 1 above)
Comment (huiran0826) [reply features/step_definitions/networking.rb]: Fixed it, thanks!

File: features/step_definitions/networking.rb (hunk 2 above)
Comment (huiran0826) [reply features/step_definitions/networking.rb]: Fixed it, thanks!

File: features/step_definitions/networking.rb (hunk 3 above)
Comment (huiran0826) [reply features/step_definitions/networking.rb:1148]: There is a class NetworkOperator in https://github.com/openshift/verification-tests/pull/1252/files, after that PR merged, I can update these codes to use that class.

File: features/step_definitions/networking.rb (hunk 4 above)
Comment (huiran0826) [reply features/step_definitions/networking.rb:1130]: There is a class NetworkOperator in https://github.com/openshift/verification-tests/pull/1252/files, after that PR merged, I can update these codes to use that class.

File: features/step_definitions/networking.rb (hunk 3 above)
Comment (huiran0826) [reply features/step_definitions/networking.rb]: @pruan-rht  Updated it, thanks!

File: features/step_definitions/networking.rb (hunk 4 above)
Comment (huiran0826) [reply features/step_definitions/networking.rb]: @pruan-rht Updated it, thanks!

**Comment by huiran0826** [comment]
Any other comments, rebase and retest, local run passed.

**Comment by huiran0826** [comment]
@zhaozhanqi @anuragthehatter @weliang1 @rbbratta  PTAL, thanks!

File: features/step_definitions/networking.rb [hunk 5]
```diff
@@ -1172,4 +1172,36 @@
...
+    cb[cb_name]="ovn-egressfirewall"
+  when "OpenShiftSDN"
+    cb[cb_name]="egressnetworkpolicy"
+  else
+    raise "unknown network_type"
+  end
+  logger.info "The egressfirewall fie directory path is stored to the #{cb_name} clipboard."
+end
+
+Given /^I get egress type to the#{OPT_SYM} clipboard$/ do | cb_name |
```
Comment (liangxia) [inline features/step_definitions/networking.rb:1193]: Generally, we use get ... from ..., get ... to ... looks strange.
Perhaps `Given /^I save egress type to the#{OPT_SYM} clipboard$/ do | cb_name |` or 
`Given /^I get egress type and save to the#{OPT_SYM} clipboard$/ do | cb_name |`

Comment (huiran0826) [reply features/step_definitions/networking.rb]: @liangxia Thanks, fixed it.

**Comment by huiran0826** [comment]
@liangxia Can you help merge if no other comments?   There is another hold PR based on this one.

File: features/step_definitions/networking.rb [hunk 6]
```diff
@@ -1180,3 +1180,35 @@
...
+  network_type = network_operator.network_type(user: admin)
+  case network_type
+  when "OVNKubernetes"
+    cb[cb_name] = "egressfirewall"
+  when "OpenShiftSDN"
+    cb[cb_name] = "egressnetworkpolicy"
+  else
+    raise "unknown network_type"
+  end
+  logger.info "The egressnetwork policy fie directory path is stored to the #{cb_name} clipboard."
```
Comment (liangxia) [inline features/step_definitions/networking.rb:1213]: It looks like most of the steps are the same, perhaps we should merge those two step definitions ?

Comment (huiran0826) [reply features/step_definitions/networking.rb:1213]: Not all the cases need both steps, some case may just need one step here, so I think it is ok to keep it as it is.

Comment (liangxia) [reply features/step_definitions/networking.rb:1213]: Then you need to update this line. It is a simple copy without modification to adapt.

Comment (huiran0826) [reply features/step_definitions/networking.rb]: aha, sorry for these typo, updated it!

**Pull request merged**